java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -1863,12 +1863,18 @@ public void setHostData(BufferEncapsulator<HostMemoryBuffer> hostData) {
       return deviceData;
     }
 
+    /**
+     * This returns total memory allocated in device for the ColumnVector.
+     * NOTE: If DType is STRING_CATEGORY, the size is estimated. The estimate assumes the length
+     * of strings to be 10 characters in each row and returns 24 bytes per row.
```

Comment:
Technically it's not 24 bytes per row but 24 bytes per dictionary entry.  Callers may be surprised that it's not just `row_count * 24`.